lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -822,23 +824,39 @@ StringRef getVerilogValueName(Value val) {
 //===----------------------------------------------------------------------===//
 
 namespace {
+void callbackOnPrint(bool open, Operation *op, unsigned line, unsigned col,
+                     StringAttr fileName) {
+  auto *ctxt = op->getContext();
+
+  mlir::LocationAttr verilogLoc =
+      mlir::FileLineColLoc::get(fileName, line + 1, col);
+  if (open && !op->hasAttr("verilogLocationBegin"))
+    op->setAttr(StringAttr::get(ctxt, "verilogLocationBegin"), verilogLoc);
+  else if (!open && !op->hasAttr("verilogLocationEnd"))
+    op->setAttr(StringAttr::get(ctxt, "verilogLocationEnd"), verilogLoc);
```

Comment:
Probably `op->setAttr` is not thread safe here since an emitter will up `hw.verilogName` from different threads.

Reply:
Yes, the stringattrs for the attr names should be setup early.  Besides the thread safety issues, there is no reason to hash strings and lookup the unique attribute every time you need to add it.